### PR TITLE
`element-event` ingestion

### PR DESCRIPTION
**Pull request by CBroz1**
- I used @tdincer's [main branch](https://github.com/tdincer/workflow-calcium-imaging) as a starting point to automatically reflect work done in PR #56 
- I copy/pasted ingestion functions from my parallel work on [workflow-array-ephys](https://github.com/cbroz1/workflow-array-ephys) - See [corresponding PR](https://github.com/datajoint/workflow-array-ephys/pull/46) for imaginary experiment structure.
- Future PRs should
   - Add a downstream table to model analysis with `element-trial`
   - Add documentation to support the present work
   - Consider migrating ingestion functions to `element-interface`, including `ingest_events()`

### CODE REVIEW

**Comment by tdincer** [comment]
Docker files are extensively tested in https://github.com/datajoint/workflow-calcium-imaging/pull/56. I suggest we rely on that PR for the conflict resolution.

**Comment by CBroz1** [comment]
> Docker files are extensively tested in #56

Agreed. I plan to resolve any conflicts on the dev branch after #56 has been merged to main. This dev PR was opened for the sake of immediacy - to given Thinh direct access to a populated version of `element-event` in this workflow

File: user_data/events.csv [hunk 1]
```diff
@@ -0,0 +1,53 @@
+subject,session_datetime,trial_id,event_id,event_start_time,event_type
```
Comment (kabilar) [inline user_data/events.csv:1]: Could you please remove the `event_id` column?

File: workflow_calcium_imaging/ingest.py [hunk 2]
```diff
@@ -57,35 +87,79 @@ def ingest_sessions(session_csv_path='./user_data/sessions.csv'):
...
+    Ingest each level of experiment heirarchy for element-trial:
+        recording, block (i.e., phases of trials), trials (repeated units),
+        events (optionally 0-duration occurances within trial).
+    This ingestion function is duplicated across wf-array-ephys and wf-calcium-imaging
+    """
+    csvs = [recording_csv_path, recording_csv_path,
+            block_csv_path, block_csv_path,
+            trial_csv_path, trial_csv_path, trial_csv_path,
+            trial_csv_path,
+            event_csv_path]
```
Comment (kabilar) [inline workflow_calcium_imaging/ingest.py:139]:
```suggestion
            event_csv_path, event_csv_path]
```

**Comment by kabilar** [comment]
Thanks @CBroz1.  Could you please pull from datajoint:main to fast forward the datajoint:dev branch with this pull request?

**Comment by kabilar** [comment]
Or close this pull request, delete the `datajoint:dev` branch, and create a new branch on the `datajoint` repository.

**Comment by CBroz1** [comment]
> Thanks @CBroz1. Could you please pull from `datajoint:main` to fast forward the `datajoint:dev` branch

All set

File: user_data/events.csv [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+subject6,2021-10-25 13:06:40,36,847.586,center
+subject7,2021-10-25 13:06:40,36,850.907,right
+subject8,2021-10-25 13:06:40,37,871.637,center
+subject9,2021-10-25 13:06:40,37,872.803,left
+subject10,2021-10-25 13:06:40,37,882.717,center
+subject11,2021-10-25 13:06:40,39,914.219,right
+subject12,2021-10-25 13:06:40,39,920.791,right
+subject13,2021-10-25 13:06:40,40,937.419,right
+subject14,2021-10-25 13:06:40,40,946.424,center
+subject15,2021-10-25 13:06:40,40,953.378,left
```
Comment (kabilar) [inline user_data/events.csv:53]: Hi @CBroz1, was adding other subjects here intentional?  If so, please ignore my comment.

File: workflow_calcium_imaging/ingest.py [hunk 4]
```diff
@@ -57,35 +87,79 @@ def ingest_sessions(session_csv_path='./user_data/sessions.csv'):
...
+                                                                         ).as_posix()})
+    new_equipment = set(val for dic in scanner_list for val in dic.values())
+    print(f'\n---- Insert {len(new_equipment)} entry(s) into experiment.Equipment ----')
     Equipment.insert(scanner_list, skip_duplicates=True)
 
     print(f'\n---- Insert {len(session_list)} entry(s) into session.Session ----')
-    session.Session.insert(session_list)
-    session.SessionDirectory.insert(session_dir_list)
+    session.Session.insert(session_list, skip_duplicates=skip_duplicates)
+    session.SessionDirectory.insert(session_dir_list, skip_duplicates=skip_duplicates)
```
Comment (kabilar) [inline workflow_calcium_imaging/ingest.py:116]:
```suggestion
    session.Session.insert(session_list)
    session.SessionDirectory.insert(session_dir_list)
```

Comment (kabilar) [reply workflow_calcium_imaging/ingest.py:116]: Not required because of the above statement:
```
if session_key not in session.Session():
```

**Review by kabilar — APPROVED** [review]
Thanks @CBroz1!  Great work.  A couple of minor suggestions above.

File: user_data/events.csv (hunk 3 above)
Comment (CBroz1) [reply user_data/events.csv]: I did not! good catch! - click/drag error with excel being 'smarter' than it should

**Comment by CBroz1** [comment]
All set - thanks for your review @kabilar

**Review by kabilar — APPROVED** [review]

**Pull request merged**